[1주차] 이경준/[feat] 프로젝트 초기 세팅 및 공통 컴포넌트 개발

KyeongJooni opened the pull request:
## ✨ 주요 변경사항
<!-- 여기에 어떤 변경을 했는지 설명해주세요-->
- 프로젝트 초기 셋팅
  - Vite + React + TypeScript, Yarn 을 활용한 개발 환경 구성
  - ESLint, Prettier, vscode 확장기능 사용하여 코드 스타일 설정
  - TailwindCSS, Autoprefixer, PostCSS를 설치 및 설정
  - Vite Alias, SVGR 플러그인, PWA 를 설정
  - axios 기반 API 통신을 위한 초기 설정
- 공통 컴포넌트 개발
---

## 📝 작업 상세 내용
<!--리뷰어가 특별히 봐줬으면 하는 부분이 있다면 여기에 적어주세요-->
- Button, Modal 등 공통 컴포넌트 구축(TextField, Divider 등)
- tailwind.config 설정을 통한 디자인 토큰 시스템 도입
- CVA(Class Variance Authority)도입으로 다양한 variant 대응
- clsx를 도입하여 유틸리티 클래스 관리
- 아이콘 시스템 정리 (iconVariants로 통일)
- alias(@) 적용으로 import 경로 일관화
- 포맷/타입 에러 정리 -> ESLint, prettier 오류 해결
- 컴포넌트 테스트를 위한 Playground 환경 도입 -> 빌드 시 빌드 안되게 변경
- 이슈 생성 및 PR 편의성을 위한 템플릿 생성
- Tailwind @apply 문법을 활용한 전역 커스텀 유틸리티 클래스 생성

---

## ✅ 체크리스트
- [x] GitHub 이슈에 `resolves: #번호`를 추가했는가?
- [x] 불필요한 주석, 디버깅 코드 제거했는가?
- [x] 기능 테스트 및 정상 동작 확인했는가?

---

## 📸 스크린샷 (선택)
<!-- UI 변경 사항이 있다면 여기에 첨부해주세요 (Drag & Drop 가능) -->
<img width="1277" height="704" alt="image" src="https://github.com/user-attachments/assets/58cb2c20-486c-40ec-90d0-a1063ea01efe" />

Playground 컴포넌트 테스트 1

<img width="1214" height="903" alt="image" src="https://github.com/user-attachments/assets/45fd7d4c-4c05-425e-90c4-b5883e60efee" />

Playground 컴포넌트 테스트 2

<img width="1194" height="756" alt="image" src="https://github.com/user-attachments/assets/23bcac9f-147f-457c-8826-5b1714f5fbe4" />

Playground 컴포넌트 테스트 3

<img width="1052" height="934" alt="image" src="https://github.com/user-attachments/assets/e3019ef5-8f31-4615-8b0c-de8a9d80640e" />

Playground 컴포넌트 테스트 4

<img width="1055" height="427" alt="image" src="https://github.com/user-attachments/assets/fce00dc1-4bd3-4995-9a2e-55392963b27d" />

Playground 컴포넌트 테스트 5

<img width="755" height="1020" alt="image" src="https://github.com/user-attachments/assets/78fa138d-10ef-4a19-9f64-1e6da4fbc01a" />

Playground 컴포넌트 테스트 6

<img width="1158" height="368" alt="image" src="https://github.com/user-attachments/assets/d6145737-6aaf-4945-ba9d-c519e57474dc" />

Playground 컴포넌트 테스트 7

<img width="564" height="282" alt="image" src="https://github.com/user-attachments/assets/167e6bfc-6755-4809-9f17-59219a261388" />

Playground 컴포넌트 테스트 8

---

## 🔍 기타 참고사항
<!--리뷰어에게 공유하고 싶은 추가 정보가 있다면 여기에 작성해주세요-->
- TailwindCSS가 익숙하지 않아서 사용방법이 비효율적일 수도 있습니다. 리뷰 달게 받겠습니다!
- 디자인 토큰/스타일 가이드 적용이 미흡합니다. 수정하겠습니다.
- 파일 구조를 임시로 잡아둔 상태라, 리팩토링 의견 적극 환영입니다!
- 커밋 컨벤션이 엉망인데, 2주차부터 신경쓰겠습니다. ㅠㅠ
---

## 🔗 관련 이슈
<!-- 반드시 관련 이슈 번호를 적어주세요. 예시: Close #123 -->
- Close #3

KyeongJooni committed:
init: 프로젝트 초기셋팅

- React, Vite, Typescript, Yarn을 기반으로 한 프로젝트 초기셋팅
- Tailwind CSS, PostCSS + Autoprefixer를 사용한 스타일링 셋팅
- ESLint, + Prettier, vscode 확장 기능 사용
- axios기반 API 통신 초기셋팅 구축

feat: 공통 컴포넌트 개발

- clsx를 활용한 유틸리티 클래스 관리
- Pagination 컴포넌트 구현 (PaginationButton, PaginationItem 분리)
- 디자인 시스템 토큰 기반 일관된 스타일링 적용
- state별 색상 관리 (default, active, disabled)
- Tailwind CSS 커스텀 토큰 확장 (character-title-85, primary 등)
- CVA를 통한 컴포넌트 variant 체계화
- TypeScript 타입 안전성 확보 (VariantProps 활용)
- 재사용 가능한 컴포넌트 아키텍처 구성

KyeongJooni committed:
mod: TailwindCSS 설정 수정

KyeongJooni committed:
mod: 컴포넌트별 ts 파일명 수정

- modal, pagination, text, textbox 수정
- main파일 절대경로 import

del: 사용하지 않는 폴더, 파일 삭제

- config폴더, env.ts 삭제

resolves: #03

KyeongJooni committed:
mod: PR템플릿 내용 수정

gachaemin commented on src/components/common/Dropdown/DropdownMenu.tsx:
```diff
@@ -0,0 +1,54 @@
+  position = 'right',
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
```
useRef + mousedown 이벤트로 dropdown을 닫는 로직을 사용할 수 있다는 점을 새로 알았어요!

KyeongJooni replied:
useRef + mousedown 조합은 dropdown뿐 아니라 모달같이 외부 클릭 시 닫혀야 하는 UI상황에 쓰면 재사용도 가능합니다!

gachaemin reviewed (commented):
개발 속도가 빠르신거같아요! 대단하십니다..!!

ld5n1z commented:
수고 너무 많으셨습니다!!!

ld5n1z commented:
질문이 있습니다! ☺️
PaginationButton.tsx과 paginationButtonVariants.ts처럼 구분하신 이유가 궁금해요!!! 더 효율적일까요??

KyeongJooni commented:
> 질문이 있습니다! ☺️ PaginationButton.tsx과 paginationButtonVariants.ts처럼 구분하신 이유가 궁금해요!!! 더 효율적일까요??

같은 파일에 두면 코드 덩어리가 커지기 때문에 분리하였고, UI 로직과 스타일 정의를 분리해서 유지보수성을 챙겨가려고 분리했습니다! TS를 쓴 이유는 cva(Class Variance Authority)는 tsx 문법을 사용하지 않기 때문에 ts를 사용했어요. 또한 tsx 같은 파일들은 번들러가 번들링 할때 변환과정을 거치기 때문에 ts같은 파일들이 더 빠르다고 알고있어서 사용해씁니다. 스타일 정의 파일이라는 것을 한눈에 알아보기도 쉽기도 하고요!

HandoA01 commented:
수고많으셨습니다! 정말 개발 속도가 빠르신 것 같아요...!!

HandoA01 commented on src/components/common/pageheader/PageHeaderRightRenderers.tsx:
```diff
@@ -0,0 +1,36 @@
+  </div>
+);
+
+// 상세 타입 렌더링
```
PageHeader 부분을 Left, Legacy 등의 파일로 구분하신 이유가 궁금합니다!
그리고 각 컴포넌트별로 Variants와 tsx로 구분하여 하신점도 새롭게 알아갑니다~

HandoA01 commented on src/components/common/pageheader/PageHeaderRightRenderers.tsx:
```diff
@@ -0,0 +1,36 @@
+      <MoreVertIcon />
+    </Icon>
+  </div>
+);
```
PageHeader 부분을 Left, Legacy 등의 파일로 구분하신 이유가 궁금합니다!
그리고 각 컴포넌트별로 Variants와 tsx로 구분하여 하신점도 새롭게 알아갑니다~

KyeongJooni replied:
피그마 디자인을 기반으로 서비스 플로우를 검토했을 때, 각 페이지마다 사용하는 PageHeader가 다르다고 판단했어요. 특히 Right 영역의 구성이 달라지는 경우가 있어 이를 분기 처리하기 위해 별도 컴포넌트로 분리했습니다!
또한 Left와 Right 영역을 하나의 파일 안에 모두 작성했을 때 코드가 불필요하게 커지고 가독성이 떨어진다고 판단하여 이를 각각 독립된 컴포넌트로 관리하고, 최종적으로는 PageHeader에서 호출해 조립하는 구조로 설계했습니다!

KyeongJooni committed:
mod: modal 버튼 스타일 수정

- border 스타일 중 색상 수정

chore: eslint config 파일 버전 업그레이드

- config 파일 이름, 내용 수정
- CR 줄바꿈 LF로 수정

theSnackOverflow commented:
공용 컴포넌트 모두 다 구현하셨네요..!
이번 기수 경준 님 코드 보면서 많이 배울 것 같습니다!

이 PR을 통해 `새로 알게 된 것` & `공부해야할 것`
- SVGR 플러그인
- PWA 
- CVA(Class Variance Authority)도입으로 다양한 variant 대응
- clsx를 도입하여 유틸리티 클래스 관리
- 아이콘 시스템 정리 (iconVariants로 통일)

ld5n1z commented:
> > 질문이 있습니다! ☺️ PaginationButton.tsx과 paginationButtonVariants.ts처럼 구분하신 이유가 궁금해요!!! 더 효율적일까요??
> 
> 같은 파일에 두면 코드 덩어리가 커지기 때문에 분리하였고, UI 로직과 스타일 정의를 분리해서 유지보수성을 챙겨가려고 분리했습니다! TS를 쓴 이유는 cva(Class Variance Authority)는 tsx 문법을 사용하지 않기 때문에 ts를 사용했어요. 또한 tsx 같은 파일들은 번들러가 번들링 할때 변환과정을 거치기 때문에 ts같은 파일들이 더 빠르다고 알고있어서 사용해씁니다. 스타일 정의 파일이라는 것을 한눈에 알아보기도 쉽기도 하고요!

CVA는 tsx 문법을 사용하지 않는군요?! 분리하니 확실히 더 보기 쉬운 것 같습니다!! 답변 감사합니다~!

theSnackOverflow commented on tsconfig.node.json:
```diff
@@ -0,0 +1,11 @@
+    "strict": true
+  },
+  "include": ["vite.config.mts"]
+}
```
`느낀 점`
이제껏 `tailwind`로 스타일링 하면서 기본적으로 제공되는 클래스만 사용하거나   color theme만 커스텀 하는 정도였는데, 이 파일 코드를 보면서 tailwindcss를 겉핥기로만 활용 하는 제 모습을 되돌아 보게 되었습니다...

theSnackOverflow commented on src/components/common/button/buttonVariants.ts:
```diff
@@ -0,0 +1,52 @@
+// 버튼 아이콘 스타일
+export const buttonIconVariants = cva('flex items-center justify-center');
+
+export type ButtonVariantProps = VariantProps<typeof buttonVariants>;
```
`질문`
저는 피그마를 보면서 버튼의 스타일 및 종류가 많아, 종류를 크게 몇 가지로 쪼개서 구현해야할 지 막막했습니다. 경준 님의 기준이 궁금합니다!

KyeongJooni replied:
기존 피그마에 있는 공통 컴포넌트 색깔 기준으로 cva로 variant를 표현하려고 했던 것 같아요. 그 대신 만약에 추가 될 수 있는 색상들도 사용할 수 있도록 하려고 했습니다!

theSnackOverflow reviewed (commented):
공용 컴포넌트 모두 다 구현하셨네요..!
이번 기수 경준 님 코드 보면서 많이 배우겠습니다!

이 PR을 통해 `새로 알게 된 것` & `공부해야할 것`
- SVGR 플러그인
- PWA 공용 컴포넌트 모두 다 구현하셨네요..!
이번 기수 경준 님 코드 보면서 많이 배우겠습니다!

이 PR을 통해 `새로 알게 된 것` & `공부해야할 것`
- SVGR 플러그인
- PWA 
- CVA(Class Variance Authority)도입으로 다양한 variant 대응
- clsx를 도입하여 유틸리티 클래스 관리
- 아이콘 시스템 정리 (iconVariants로 통일)
- CVA(Class Variance Authority)도입으로 다양한 variant 대응
- clsx를 도입하여 유틸리티 클래스 관리
- 아이콘 시스템 정리 (iconVariants로 통일)

KyeongJooni committed:
mod: 페이지헤더 스타일 수정

eileen4505 commented on src/components/common/button/Button.tsx:
```diff
@@ -0,0 +1,35 @@
+    <button
+      type={type}
+      onClick={onClick}
+      className={clsx(buttonVariants({ intent, size, variant, fullWidth }), className)}
```
`buttonVariantf` 결과를 `clsx`로 합치는 방식이 깔끔해서 좋은 것 같아요!

eileen4505 commented on src/components/common/button/buttonVariants.ts:
```diff
@@ -0,0 +1,52 @@
+      fullWidth: false,
+    },
+  }
+);
```
cva로 intent/size/variant를 나눠둔 방식이 옵선별로 잘 정리돼 있어서 읽기 편하고 깔금한 것 같습니다!!배워갑니다:)

eileen4505 reviewed (commented):
짧은 시간동안 정말 빠르고 깔끔하게 구현하신 것 같아요👏🏻 많이 배워갑니다!!

dalzzy commented on src/api/auth/auth.ts:
```diff
@@ -0,0 +1,90 @@
+  }
+};
+
+// 로그인 API
+export const login = async (loginData: LoginRequest): Promise<LoginResponse> => {
+  const response = await axiosInstance.post<LoginResponse>('/auth/login', loginData);
+
+  const { accessToken, refreshToken } = response.data.data;
+
+  // 로그인 성공 시 토큰들 저장
+  setAccessToken(accessToken);
+  setRefreshToken(refreshToken);
+
+  return response.data;
+};
+
+// 회원가입 API
+export const register = async (registerData: RegisterRequest): Promise<RegisterResponse> => {
+  const response = await axiosInstance.post<RegisterResponse>('/auth/register', registerData);
+  return response.data;
+};
+
+// OAuth2 회원가입 API
+export const registerOAuth = async (registerData: RegisterOAuthRequest): Promise<RegisterOAuthResponse> => {
+  const response = await axiosInstance.post<RegisterOAuthResponse>('/auth/register-oauth', registerData);
+  return response.data;
+};
+
+// OAuth2 로그인 리다이렉트 URL 조회
+export const getKakaoRedirectUrl = async (): Promise<KakaoRedirectResponse> => {
+  const response = await axiosInstance.get<KakaoRedirectResponse>('/auth/kakao');
+  return response.data;
+};
+
+// OAuth2 로그인 콜백 처리
+export const handleKakaoRedirect = async (
+  callbackData: KakaoRedirectRequest
+): Promise<KakaoRedirectCallbackResponse> => {
+  const response = await axiosInstance.get<KakaoRedirectCallbackResponse>('/auth/kakao/redirect', {
```
api 코드들도.. 벌써 작성해두셨네요..ㄷㄷ 따봉드립니다..

KyeongJooni replied:
그냥 구조만 가져온거라서 나중에 수정해서 사용할 예정입니다!

dalzzy commented on src/api/auth/types.ts:
```diff
@@ -0,0 +1,93 @@
+  nickname?: string;
+}
+
+export interface RegisterOAuthResponse {
+  code: number;
+  message: string;
+  data: {
+    accessToken: string;
+    refreshToken: string;
+    nickname: string;
+    profilePicture: string;
+    introduction: string;
+    httpStatus: string;
+    responseMessage: string;
+  };
+}
+
+// 카카오 리다이렉트 URL 응답
+export interface KakaoRedirectResponse {
+  code: number;
+  message: string;
+  data: {
+    redirectUrl: string;
+  };
+}
+
+// 카카오 콜백 요청/응답
+export interface KakaoRedirectRequest {
+  code: string;
+  state?: string;
+}
+
+export interface KakaoRedirectCallbackResponse {
+  code: number;
+  message: string;
+  data: {
+    accessToken: string;
```
현재 응답들이 code / message / data 구조를 공통으로 사용하고 있어서,


```
export interface ApiResponse<T> {
  code: number ;
  message: string;
  data: T;
}
```
이런 식으로 공통 응답 타입을 제네릭으로 분리해 두면 유지보수에 더 좋을 것 같은데 어떻게 생각하시나용??

KyeongJooni replied:
제네릭 타입을 사용을 잘 안해봐서 interface로 하나하나 구현했는데 타입의 재사용성을 높이는 방법 적용해보겠습니다!

dalzzy commented on src/components/common/modal/Modal2.tsx:
Modal파일과 Modal2파일은 코드가 동일해보이는데 어떤 차이가 있는걸까요...?

KyeongJooni replied:
피그마 디자인 상으로 모달이 2개 있길래 밤늦게 생각없이 만들다가 중복된 코드가 작성된 거 같아요 ㅠㅠ 수정하겠습니다!

dalzzy commented on src/components/common/modal/Modal.tsx:
```diff
@@ -0,0 +1,45 @@
+  modalDeleteButtonVariants,
+} from '@/components/common/modal/ModalVariants';
+import clsx from 'clsx';
+import React from 'react';
```
`import React from 'react';` 문은 React 17버전 이후부터 명시하지 않아도 되는 걸로 알고 있는데, 
경준님 개발 환경에서는 이 import문이 필요한걸까요?? 궁금해서 여쭤봅니다..!

KyeongJooni replied:
usestate나 리액트 훅을 가져올 때 react에서 가져오기 때문에 보편적으로 초기 개발단계에서 저렇게 적어놓고 나중에 수정해서 사용하는 편입니다!. 다만 필요없는걸 알게되었어도 기존에 리액트를 공부할 때 써 버릇해서 적은 것 같습니다... 수정해씁니다!

dalzzy replied:
아하 그렇군요!! 굿굿

dalzzy commented on src/components/common/modal/Modal2.tsx:
```diff
@@ -0,0 +1,45 @@
+        </div>
+      </div>
+    </div>
+  );
```
현재는 취소,삭제하기로 모달 내부의 텍스트가 고정되어 있는데 피그마 확인해보시면 
<img width="414" height="236" alt="image" src="https://github.com/user-attachments/assets/10c62c3a-18a1-4e43-ac58-6ae3b37051d7" />
버튼의 색상 및 텍스트가 다른 모달들도 있어 현재는 재사용성이 쪼금 떨어질 수 있을 것 같습니다!!

버튼의 텍스트와 색상을 props로 받는 형태로 수정해보면 어떨까요??

KyeongJooni replied:
cva 사용하여 버튼 및 컨테이너 스타일 관리하고 구조분해 할당으로 props 받도록 수정하였습니다!

dalzzy commented on src/components/common/modal/modalVariants.ts:
```diff
@@ -0,0 +1,32 @@
+  'flex w-[326px] flex-col items-start gap-6 rounded border-0 p-6 pb-4 bg-white shadow-modal'
+);
+
+// 모달 콘텐츠 컨테이너 스타일
+export const modalContentVariants = cva('flex flex-col items-start gap-2 self-stretch rounded-xl', {
+  variants: {
+    padding: {
+      default: 'p-1',
+      none: 'p-0',
+    },
+  },
+  defaultVariants: {
+    padding: 'default',
+  },
+});
+
+// 모달 버튼 컨테이너 스타일
+export const modalButtonsVariants = cva('flex items-center gap-3 self-stretch');
+
+// 모달 취소 버튼 스타일
+export const modalCancelButtonVariants = cva(
+  'flex px-3 py-2 justify-center items-center gap-2 flex-1 text-sm font-normal text-black text-right rounded-sm border border-gray-96'
+);
+
+// 모달 삭제 버튼 스타일
+export const modalDeleteButtonVariants = cva(
+  'flex px-3 py-2 justify-center items-center gap-2 flex-1 rounded-sm text-sm font-normal border-0 cursor-pointer text-white bg-warning hover:bg-warning/90'
```
모달 취소/삭제 버튼 스타일이 같은 스타일임에도 클래스 순서가 달라 보여서, Prettier의 Tailwind 플러그인(`prettier-plugin-tailwindcss`)설치 후 설정해서 쓰면 어떨지 제안드려 봅니다!
별 건 아니지만,, 이 플러그인을 적용하면 Tailwind 클래스가 자동으로 일관된 규칙에 맞게 정렬돼서, 가독성이 좋아지는 것 같애용..

KyeongJooni replied:
좋은 정보 감사합니다. 수정했습니다!

dalzzy commented on src/assets/icons/common/profile_1.svg:
아이콘 컴포넌트를 따로 구현해주신 것 같은데 기본 profile  svg를 크기별로 저장해두신 이유가 있으신가용??

KyeongJooni replied:
아이콘 컴포넌트를 사용 안하고 빠른 개발을 위해 프로필 사진 svg를 설정하여 구현하였는데, 추후에 UI 개발을 할 때 같이 수정해보도록 하겠습니다!

dalzzy commented on src/components/common/pageheader/PageHeaderRight.tsx:
`pageheader`폴더명은 카멜케이스로 통일하지 않으신 이유가 있으신걸까용??

KyeongJooni replied:
기존에 PR 이후에 확인하여 수정 후 다시 push하였는데 깃허브에서 파일명 대소문자 변경은 인식을 못하는 경우가 있더라고요...

dalzzy replied:
아하 마자요 .... 멍청한 깃허브.. 다른 이름으로 변경하고 커밋했다가 다시 파일명 변경해야 반영되더라구요 ..

dalzzy commented on src/components/common/icon/IconVariants.ts:
```diff
@@ -0,0 +1,44 @@
+import { cva } from 'class-variance-authority';
```
이런 라이브러리도 있군요,, 좋은 거 알아갑니당👍🏻

dalzzy commented on src/layout/Sidebar.tsx:
```diff
@@ -0,0 +1,67 @@
+import { Profile1Icon } from '@/assets/icons/common';
+import Button from '@/components/common/button/Button';
+import Spacer from '@/components/common/spacer/Spacer';
```
components폴더 안에도 index파일을 따로 만들어서 모든 컴포넌트들을 export해주면 
`import {Button,Spacer} from "@/components"` 이런 식으로  import문이 더 깔끔해질 것 같애옹

KyeongJooni replied:
넵 참고하여 barrel 파일 생성하여 수정했습니다!

dalzzy commented on src/components/common/Dropdown/DropdownMenu.tsx:
```diff
@@ -0,0 +1,54 @@
+import clsx from 'clsx';
```
지금처럼 clsx 라이브러리만 써도 충분하지만 !! 이외에도 중복되는 Tailwind 클래스 속성을 제거해주는 `tailwind-merge` 라이브러리도 있는데, 이 두 개 라이브러리를 하나의 유틸함수로 분리해서 사용하는 걸 고려해보시면 어떨까용??

```
import clsx, { type ClassValue } from 'clsx';
import { twMerge } from 'tailwind-merge';

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

```
꼭 이렇게 써야 하는 건 아니지만 보통 이렇게 많이들 사용하시더라구요 . . . 
이렇게 분리하고 기존 clsx(...) 자리에 cn(...)만 사용하시면 됩니다!

KyeongJooni replied:
넵 수정완료했습니다! 기존에 clsx를 사용한 후에 tailwind-merge 라이브러리가 있다는 것을 알게되어서 적용하는 것에 대해 고민을 했었는데 사용하니까 확실히 가독성이 좋아진 것 같습니다👍

dalzzy commented on src/components/common/pagination/Pagination.tsx:
```diff
@@ -0,0 +1,21 @@
+    <div className={clsx('inline-flex items-start gap-2', className)}>
+      <PaginationButton variant="navigation" direction="prev" />
+
+      <PaginationButton variant="page">1</PaginationButton>
```
페이지 번호가 1로 고정돼 보이는데, 임시 값일까요?? 추후에 어떻게 활용하실 예정인지 궁금합니당..

KyeongJooni replied:
맞습니다. 현재는 하드코딩으로 임시 값을 설정하였고, UI 컴포넌트의 기본 구조와 스타일링을 먼저 완성하는 단계로, 실제 페이지네이션 로직은 추후에 구현할 예정입니다.
Props 인터페이스 확장과 현재 페이지 상태와 총 페이지 수를 관리하는 로직을 구현하고 백엔드에서 보내는 페이지 갯수에 맞춰 페이지를 그룹핑하려고 합니다!

dalzzy replied:
아항 답변 감사합니다~!!

dalzzy commented on src/components/common/toast/ToastPositive.tsx:
ToastPositive/ToastWarning으로 나누니까 의도가 분명해보여서 좋은 것 같아요!!
다만 두 컴포넌트의 스타일이 거의 유사한 것 같은데, 하나의 Toast 컴포넌트로 합치고 토스트 타입(`positive | warning`)에 따라 아이콘/색상만 바꾸는 구조도 괜찮을 것 같은데 어떻게 생각하시나요??

KyeongJooni replied:
분기처리를 하기 귀찮음이 여기서 들통나버렸네요... 수정완료했습니다!

dalzzy replied:
ㅋㅋㅋㅋㅋㅋㅋ 구웃

dalzzy reviewed (commented):
수고하셨습니다~~👍🏻 개발 속도가 진짜 빠르시네요,, 덕분에 저도 코드 보면서 많이 배웠습니당 
코멘트 확인해주시고 의견 남겨주세요!!! 파이팅입니다..

KyeongJooni committed:
mod: 불필요한 import React 구문 제거

KyeongJooni committed:
mod: auth타입저의 제네릭으로 분리

- 공통 API 응답 구조를 제네릭을 분리하여 재사용성 높임

rename: Types 파일명을 authTypes로 재설정

KyeongJooni committed:
mod: 모달 버튼과 텍스트 색상 props 방식으로 수정

- ModalVariants를 적용하여 버튼 및 컨테이너 스타일 관리

KyeongJooni committed:
mod: 테일윈드 플러그인 설정

- 프로젝트 전체 코드 일관성있게 수정

KyeongJooni committed:
mod: 사이드바 import 구문 간소화

- components에 barrel 파일을 만들어 export

KyeongJooni committed:
mod: Toast 컴포넌트 구조 변경

- 기존 분리되어있던 2개의 파일을 cva로 통합하여 관리

mod: tailwind-merge 라이브러리 적용

- clsx를 사용하던 유틸 클래스를 tailwind-merge 라이브러리 활용하여 스타일 중복 제거

dalzzy reviewed (approved):
수정사항 확인했습니다~ 수고하셨습니다 👍🏻👍🏻

KyeongJooni committed:
rename: 컴포넌트 폴더명 변경

- 파스칼 케이스로 통일

KyeongJooni committed:
fix: eslint 오류 제거
mod: PageHeaderLagacy내 텍스트 수정
